lib/src/cardano_ledger_application.dart

```diff
@@ -284,18 +274,15 @@ class CardanoLedgerApp {
     int addressIndex = 0,
     bool displayOnDevice = false,
   }) async {
-    final bip32StakePath = [
-      harden + 1852,
-      harden + 1815,
-      harden + accountIndex,
-      0,
-      addressIndex,
-    ];
+    final path = LedgerSigningPath.shelley(
+      account: accountIndex,
+      address: addressIndex,
+      role: ShelleyAddressRole.stake,
```

Comment:
The role here is wrong. This should be payment role for `deriveEnterpriseAddress`

Reply:
Also please make sure to add 1-2 tests for this to catch such a mistake in the future.